org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -1454,15 +1454,40 @@ public Preferences setGradleUserHome(String gradleUserHome) {
 	}
 
 	public Preferences setFormatterUrl(String formatterUrl) {
-		this.formatterUrl = ResourceUtils.expandPath(formatterUrl);
+		this.formatterUrl = this.resolveWorkspaceFolder(ResourceUtils.expandPath(formatterUrl));
 		return this;
 	}
 
 	public Preferences setSettingsUrl(String settingsUrl) {
-		this.settingsUrl = ResourceUtils.expandPath(settingsUrl);
+		this.settingsUrl = this.resolveWorkspaceFolder(ResourceUtils.expandPath(settingsUrl));
 		return this;
 	}
 
+	private String resolveWorkspaceFolder(String filePath) {
+		final String workspaceFolderVariable = "${workspaceFolder}";
```

Comment:
I would use `WORKSPACE_FOLDER_VAR` just to make it a bit more obvious that it's constant.